src/app/sitemap.ts

```diff
@@ -0,0 +1,85 @@
+import type { MetadataRoute } from "next";
+import { getBlogPosts } from "@/services/api";
+import { cookies } from "next/headers";
+
+export default async function sitemap(): Promise<MetadataRoute.Sitemap> {
+  cookies()
+  const now = Date.now();
+  // Filter non-archived blog posts
+  const blogPosts: MetadataRoute.Sitemap = (await getBlogPosts())
+    .filter((b) => {
+      if (b.end_at !== null && b.end_at !== undefined) {
+        const endDate = Date.parse(b.end_at);
+        return now < endDate;
+      } else {
+        return true;
+      }
+    })
+    .map((blog) => ({
+      url: `https://www.now-u.com/blog/${blog.slug}`,
+      lastModified: new Date(blog.release_at),
```

Comment:
The updated date is already available, yeah it may be worth to use that instead but you already merged it so... I don't think it matters a lot atm though.